spec/lib/miq_automation_engine/models/miq_ae_domain_version_spec.rb

```diff
@@ -0,0 +1,66 @@
+require "spec_helper"
+include AutomationSpecHelper
+
+describe MiqAeDomain do
+  before do
+    EvmSpecHelper.local_guid_miq_server_zone
+    @user = FactoryGirl.create(:user_with_group)
+    @temp_dir = Dir.mktmpdir
+    require 'miq_ae_datastore'
+    stub_const("MiqAeDatastore::DATASTORE_DIRECTORY", @temp_dir)
+  end
+
+  after do
+    FileUtils.remove_entry_secure(@temp_dir) if Dir.exist?(@temp_dir)
+  end
+
+  context "version" do
+    it "no about class" do
+      d1 = FactoryGirl.create(:miq_ae_domain, :name => 'Obelix')
+      expect(d1.version).to be_nil
+    end
+
+    it "about class present" do
+      ver = '3.0.9'
+      expect(create_about_class('Dogmatix', ver).version).to eq(ver)
+    end
+  end
+
+  context "available_version" do
+    it "no version specified" do
+      d1 = FactoryGirl.create(:miq_ae_domain, :name => 'Asterix')
+      expect(d1.available_version).to be_nil
+    end
+
+    it "about file present" do
+      version = '7.1.1'
+      dom2 = create_about_class('Vitalstatistix', version)
+      write_about_file(dom2, version)
+      expect(dom2.available_version).to eq(version)
+    end
+  end
+
+  def write_about_file(domain, version)
+    schema = [{"field" => {"name" => "version", "default_value" => version}}]
+    hash = {"object_type" => "class", "version" => 1.0,
+            "object" => {"attributes" => {"name" => "About"}, "schema" => schema}}
+    fname = File.join(MiqAeDatastore::DATASTORE_DIRECTORY, domain.name.downcase, 'system',
+                      "about#{MiqAeYamlImportExportMixin::CLASS_DIR_SUFFIX}",
+                      MiqAeYamlImportExportMixin::CLASS_YAML_FILENAME)
+    dirname = File.dirname(fname)
+    FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
```

Comment:
You do not need the `File.directory?` check.
That would also allow you to drop the `dirname` variable: `FileUtils.mkdir_p(File.dirname(fname))`